tools/swift-inspect/Sources/swift-inspect/Process.swift

```diff
@@ -43,6 +43,59 @@ private func refersToSelf(_ str: String) -> Bool {
   // No match.
   return false
 }
+
+internal func getRemoteProcess(processId: ProcessIdentifier,
+                               options: UniversalOptions) -> (any RemoteProcess)? {
+  return DarwinRemoteProcess(processId: processId,
+                             forkCorpse: options.forkCorpse)
+}
+
+internal func getProcessName(processId: ProcessIdentifier) -> String? {
+  var info = proc_bsdinfo()
+  let bsdinfoSize = Int32(MemoryLayout<proc_bsdinfo>.stride)
+  let size = proc_pidinfo(processId, PROC_PIDTBSDINFO, 0, &info, bsdinfoSize)
+  if (size != bsdinfoSize) {
+    return nil
+  }
+  let processName = withUnsafePointer(to: info.pbi_name) {
+    $0.withMemoryRebound(to: CChar.self, capacity: MemoryLayout<CChar>.stride) {
+      String(cString: $0)
```

Comment:
Is this name field guaranteed to be NUL terminated, or can it contain a name that fills the entire array with no terminator? It looks like it can probably fill the entire array, so this can overflow. We can address this and make the code a little nicer by using an unsafe buffer instead of an unsafe pointer:

```
  let processName = withUnsafeBytes(of: info.pbi_name) { buffer in
    let nonnullBuffer = buffer.prefix { $0 != 0 }
    return String(decoding: nonnullBuffer, as: UTF8.self)
  }
```

Reply:
It is guaranteed to be NUL terminated.